### PR TITLE
[SPARK-18849][ML][SPARKR][DOC] vignettes final check reorg

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Reorganizing content (copy/paste)

## How was this patch tested?

https://felixcheung.github.io/sparkr-vignettes.html

Previous:
https://felixcheung.github.io/sparkr-vignettes_old.html

### CODE REVIEW

**Comment by felixcheung** [comment]
@mengxr @shivaram

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 1]
```diff
@@ -447,31 +447,43 @@ head(teenagers)
...
 * Latent Dirichlet Allocation (LDA)
 
-* Logistic Regression Model
+#### Collaborative Filtering
 
-* Multilayer Perceptron Model
+* Alternating Least Squares (ALS)
 
-* Naive Bayes Model
+#### Test
```
Comment (yanboliang) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: Should ```Statistics``` be better, to follow MLlib organization way?

**Comment by felixcheung** [comment]
Sure? I got that from the Scala API namespace. There might be more test stuff we could add in the future?

**Comment by yanboliang** [comment]
Yeah, ```org.apache.spark.mllib.stat.test.KolmogorovSmirnovTest``` is the path, however, ```stat``` is the same level with ```classification, regression, clustering, etc``` in the directory structure. ```test``` is the sub-level compared with others. Thanks.

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 2]
```diff
@@ -496,9 +508,114 @@ count(carsDF_test)
...
+
+Multinomial logistic regression against three classes
+```{r, warning=FALSE}
+df <- createDataFrame(iris)
+# Note in this case, Spark infers it is multinomial logistic regression, so family = "multinomial" is optional.
+model <- spark.logit(df, Species ~ ., regParam = 0.056)
+summary(model)
+```
+
+#### Multilayer Perceptron
```
Comment (yanboliang) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: Nit: ```Multilayer Perceptron``` -> ```Multilayer Perceptron Classifier (MLPC)```

**Review by yanboliang — COMMENTED** [review]
One minor question: Should we make the name of algorithms or models consistent? For example, ends with ```model``` or not. FYI, the SparkR user guide MLlib session:
image

File: R/pkg/vignettes/sparkr-vignettes.Rmd (hunk 2 above)
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: I think `MLP` is more commonly used in R.

re: the other question, I think we should try to stick with the convention in R and not add "model" in the names of everyone

Comment (mengxr) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: +1 on removing `model` at the end

Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: since that's the programming guide, feel free to take on updating them @yanboliang or I'll get around later.

Comment (yanboliang) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: @felixcheung I found we need to update this PR for [here](https://github.com/apache/spark/pull/16301/files#diff-56790892076daac0bc43a9b38c0b1470R452) and some other places, so feel free to make change for user guide as well in your update. Thanks.

**Comment by yanboliang** [comment]
LGTM

**Comment by felixcheung** [comment]
merged to master and branch-2.1, thanks

**Pull request closed without merging**